provider/virtio_fs_utils.py

```diff
@@ -0,0 +1,232 @@
+            error_context.context("Find the %s on the host." %
+                                  (host_dir + "/" + folder_name + "/a"),
+                                  LOG_JOB.info)
+        else:
+            LOG_JOB.error("Do NOT find the sub folder on the host.")
+    finally:
+        error_context.context("Delete the sub folder on shared directory "
+                              "of guest: ", LOG_JOB.info)
+        if os_type == "linux":
+            session.cmd("rm -rf %s" % (guest_dest + "/" + folder_name))
+        else:
+            session.cmd("rmdir /s/q %s" % (guest_dest + "\\" + folder_name))
+
+
+def install_psexec(vm):
+    """
+    Copying psexec.exe from host to guest.( Windows only )
+
+    :param vm: the vm object
+    :return psexec_path: the psexec path in guest
+    """
+    src_file = os.path.join(data_dir.get_deps_dir("psexec"), "PsExec.exe")
+    dest_dir = "%systemdrive%\\"
+    vm.copy_files_to(src_file, dest_dir)
+    return os.path.join(dest_dir, "PsExec.exe")
+
+
+def basic_io_test_via_psexec(test, params, vm, usernm, pwd):
+    """
+    Use psexec to do the io test( Windows only ).
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param vm: the vm object
+    :param usernm: the username used to execute the cmd
+    :param pwd: the password used to execute the cmd
+    """
+    if params.get("os_type", "windows") == "windows":
+        error_context.context("Running viofs basic io test via psexec",
+                              LOG_JOB.info)
+        cmd_dd_win = params.get("virtio_fs_cmd_dd_win",
+                                "dd if=/dev/random of=%s bs=1M count=100")
+        test_file = params.get('virtio_fs_test_file', "virtio_fs_test_file")
+        io_timeout = params.get_numeric("fs_io_timeout", 120)
+        fs_source = params.get("fs_source_dir", "virtio_fs_test/")
+        fs_target = params.get("fs_target", "myfs")
+        base_dir = params.get("fs_source_base_dir",
+                              data_dir.get_data_dir())
+        if not os.path.isabs(fs_source):
+            fs_source = os.path.join(base_dir, fs_source)
+        host_data = os.path.join(fs_source, test_file)
+
+        session = vm.wait_for_login()
+        driver_letter = get_virtiofs_driver_letter(test, fs_target, session)
+        fs_dest = "%s:" % driver_letter
+        guest_file = os.path.join(fs_dest, test_file)
+        cmd_io_test = "%systemdrive%\\cmd_io_test.bat"
+
+        error_context.context("Creating the test file(cmd_io_test.bat) "
+                              "on guest", LOG_JOB.info)
+        session.cmd("echo " + cmd_dd_win % guest_file + " > " + cmd_io_test,
+                    io_timeout)
+
+        psexec_path = install_psexec(vm)
+        try:
+            error_context.context("Execute the cmd_io_test.bat on guest",
+                                  LOG_JOB.info)
+            dns = params.get("dns", "")
```

Comment:
dns name isn't configured in cfg. 
It should be changed to domain_dns, like the main function. Thanks.